src/mcts/node.cc

```diff
@@ -113,9 +113,50 @@ Move Edge::GetMove(bool as_opponent) const {
   return m;
 }
 
+// We use strictly 16 bits to store policy values in memory-critical Edges.
```

Comment:
```cpp

// Priors (P) are stored in compressed 16-bit format.

// Source values are 32-bit floats:
// * bit 31 is sign (zero means positive)
// * bit 30 is sign of exponent (zero means negative)
// * bits 29..23 are value bits of exponent
// * bits 22..0 are mantissa bits
// (see https://www.h-schmidt.net/FloatConverter/IEEE754.html for details).

// In compressed 16-bit value we store bits 27..12:
// * bit 31 is always 0 as values are always >= 0
// * bit 30 is always 0 as values are always < 2
// * bits 29..28 are only 0 for values < 4.65e-10, assume they are always 1
// * bits 11..0 are for higher precision, they are droppes leaving only 11 bits
//              of precision.

// When converting to compressed format, bit 11 is added to make it
// a rounding rather than truncation.

// Out of 65556 possible values, 2047 are outside of [0;1] interval (they are in
// interval (1;2)).
```

Reply:
Excellent comment!